PowerEditor/src/Utf8_16.cpp

```diff
@@ -171,7 +171,10 @@ size_t Utf8_16_Read::convert(char* buf, size_t len)
 					delete [] m_pNewBuf;
                 m_pNewBuf  = NULL;
                 m_pNewBuf  = new ubyte[newSize];
-				m_nAllocatedBufSize = newSize;
+				if (m_pNewBuf)

```

Comment:
pCur could be NULL.
Rather throw std::bad_alloc as m_pNewBuf get NULL from new operator.